### DISCUSSION TITLE
i have problem with allure-results it's not updated only the allure-reports is updated

### DISCUSSION THREAD

**Discussion by Eyatester** [cypress-io/cypress#28844] [Questions and Help]
i use cypress 12.8.1     "@shelex/cypress-allure-plugin": "^2.40.1",
this is the cypress.report.json file content const report = require('multiple-cucumber-html-reporter');
const dayjs = require('dayjs');
const fs = require('fs');

const data = fs.readFileSync('cypress/reports/results.json', { encoding: 'utf8', flag: 'r' });
const runInfo = JSON.parse(data);

const osName = () => {
  switch (runInfo['osName']) {
    case 'darwin':
      return 'osx';
    case 'win32':
      return 'windows';
    case 'ubuntu':
      return 'ubuntu';
    default:
      console.log('Undefined browser');
  }
};

report.generate({
  jsonDir: 'cypress/reports/json',
  reportPath: 'cypress/reports',
  metadata: {
    browser: {
      name: runInfo['browserName'],
      version: runInfo['browserVersion'],
    },
    device: 'Local Test Machine',
    platform: {
      name: osName(),
      version: runInfo['osVersion'],
    },
  },
  customData: {
    title: 'Run Info',
    data: [
      { label: 'Project', value: 'Sample Project' },
      { label: 'Release', value: '1.0.0' },
      { label: 'Cypress Version', value: runInfo['cypressVersion'] },
      { label: 'Node Version', value: runInfo['nodeVersion'] },
      {
        label: 'Execution Start Time',
        value: dayjs(runInfo['startedTestsAt']).format('YYYY-MM-DD HH:mm:ss.SSS'),
      },
      {
        label: 'Execution End Time',
        value: dayjs(runInfo['endedTestsAt']).format('YYYY-MM-DD HH:mm:ss.SSS'),
      },
    ],
  },
  disableLog: true,
  pageTitle: 'Cypress Cucumber Html Report Example',
  openReportInBrowser: true,
  displayDuration: true,
});
 please help me to find a solution it's blocked on a past date 
![image](https://github.com/cypress-io/cypress/assets/150019045/fb13886d-32bf-4d06-986a-77f3b0f7b7ca)

**Comment by portyu9** [comment]
Hi @Eyatester 

The code you posted is not configuring Allure. It is configuring `multiple-cucumber-html-reporter`, which is a separate reporting pipeline.

With `@shelex/cypress-allure-plugin`, make sure the Allure writer is registered in `setupNodeEvents`:

```js
const { defineConfig } = require('cypress')
const allureWriter = require('@shelex/cypress-allure-plugin/writer')

module.exports = defineConfig({
  e2e: {
    setupNodeEvents(on, config) {
      allureWriter(on, config)
      return config
    },
  },
})
```

and import the plugin from your Cypress support file:

```js
import '@shelex/cypress-allure-plugin'
```

Then run Cypress with Allure enabled:

```bash
npx cypress run --env allure=true
```

The important distinction is:

- `allure-results` contains the raw results written by the Cypress adapter.
- `allure-report` is generated later from those results.
- Your `multiple-cucumber-html-reporter` configuration does not update `allure-results`.

I would first delete the old `allure-results` and `allure-report` directories, run Cypress again with `--env allure=true`, and confirm that new files appear in `allure-results` **before** generating the HTML report.

If you have multiple plugins registering `after:spec` handlers, also check the Shelex plugin's `allureReuseAfterSpec` option because competing handlers can affect result generation.

For a project being upgraded today, the newer `allure-cypress` adapter is another option, but it requires a newer Cypress version than the `12.8.1` shown in this discussion. If you remain on Cypress 12.8.1, configure the Shelex plugin correctly rather than mixing the two adapters.

Please mark this answer as accepted if it helped, thank you.
